test/cli-tests/no_permission_warn_test.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+. ./test/helper.sh
+
+test_install_dir="$test_fixtures_dir/no_permission_warn_test"
+
+function setUp()
+{
+	mkdir -p $test_install_dir
+	chmod -w $test_install_dir
+}
+
+function test_no_install_when_write_permissions_lacking()
+{
+	local output="$(ruby-install --install-dir "$test_install_dir" ruby)"
+
+	assertEquals "did not return 0" 0 $?
```

Comment:
Should test that `$?` is `1` since we're exiting with `1`.

Reply:
I think there's an error with the test when exiting with 1 status

Reply:
Try doing:

```shell
local ret
local output="$(...)" || ret=$?

assertEquals "..." -1 $ret
```

That should safely capture the exit status.

Reply:
this would return the exit status of setting the variable `local output`, if `ret` is not declared before, and if it was declared with `local ret` it will print hold an `empty value`, the solution is a little tricky, example:
```bash
function err(){                                                                           
  echo "bye"                                                                              
  exit 3                                                                                  
}                                                                                       
                                                                                             
function setting(){                                                                       
  local output                                                                            
  local e                                                                                 
  #Z5Z5Z is for uniqness                                                                  
  inter="$(err)Z5Z5Z$?"                                                                   
  output="$(echo $inter | awk 'BEGIN { FS="Z5Z5Z" } { print $1 }')"                       
       e="$(echo $inter | awk 'BEGIN { FS="Z5Z5Z" } { print $2 }')"                       
    echo $output  #will print "bye"                                                                            
    echo $e       #will print 3 (the exit status of err() function)                                                                                 
 }
```

Reply:
also the fail logging function exit with status -1, but in Unix the status codes are unsigned 8-bit integers between 0 and 255, so `exit -1` becomes `exit 255`, should I open new issue for it?

Reply:
Ah, you are correct. Yes compare against `255`.

Reply:
I've gone with  `error()` instead (checkout last commit), I think it's better to change the fail() exit status so it become obvious what status is exiting with.